nion/swift/ExportDialog.py

```diff
@@ -263,20 +263,38 @@ class UnitType(enum.Enum):
 class ExportSizeModel(Observable.Observable):
     def __init__(self, display_item: DisplayItem.DisplayItem) -> None:
         super().__init__()
-        display_size = self.__calculate_display_size_in_pixels(display_item)
-        self.__width = display_size.width
-        self.__height = display_size.height
-        self.__aspect_ratio = self.__width / self.__height
-        self.__units = UnitType.PIXELS
-        self.__float_to_string_converter = Converter.FloatToStringConverter()
-        self.__primary_field = 'width'  # Primary field to determine which text is calculated
-        self.__enforce_width_height_constraints()
+        if display_item.data_item is not None:
+            self.__image_2d = display_item.data_item.is_data_2d
+            self.__display_item = display_item
+            display_size = self.__calculate_display_size_in_pixels(display_item)
+            self.__width = display_size.width
+            self.__height = display_size.height
+            self.__aspect_ratio = self.__width / self.__height
+            self.__units = UnitType.PIXELS
+            self.__float_to_string_converter = Converter.FloatToStringConverter()
+            self.__primary_field = 'width'  # Primary field to determine which text is calculated
 
     def __calculate_display_size_in_pixels(self, display_item: DisplayItem.DisplayItem) -> Geometry.IntSize:
-        if display_item.display_data_shape and len(display_item.display_data_shape) == 2:
+        if display_item.display_data_shape and self.__image_2d:
             return Geometry.IntSize(height=display_item.display_data_shape[0], width=display_item.display_data_shape[1])
         return Geometry.IntSize(height=288, width=480)
 
+    @property
+    def image_info(self) -> typing.Optional[str]:
+        assert self.__display_item.data_item and self.__display_item.dimensional_shape
```

Comment:
This assert prevents exporting a composite line plot (a line plot with multiple layers).